src/contexts/registration/RegistrationContext.js

```diff
@@ -0,0 +1,52 @@
+import React, { createContext, useState } from 'react';
+
+const RegistrationContext = createContext();
+
+export const RegistrationState = props => {
+  const [user, setUser] = useState({
+    email: '',
+    password: '',
+    passwordRepeat: ''
+  });
+
+  const [tooltip, setTooltip] = useState({
+    emailTooltip: false,
+    passwordTooltip: false,
+    passwordRepeatTooltip: false
+  });
+
+  const [termsAccepted, setTermsAccepted] = useState(false);
+
+  const { email, password, passwordRepeat } = user;
+  const { emailTooltip, passwordTooltip, passwordRepeatTooltip } = tooltip;
+
+  const onTextChanged = e => {
+    setTooltips(e.target);
+    setUser({ ...user, [e.target.id]: e.target.value });
+  };
+
+  const setTooltips = target => {
+    if (target.validity.patternMismatch || target.validity.typeMismatch) {
+      setTooltip({ ...tooltip, [target.id + 'Tooltip']: true });
+    } else setTooltip({ ...tooltip, [target.id + 'Tooltip']: false });
+  };
+
+  return (
+    <RegistrationContext.Provider
+      value={{
+        email,
+        password,
+        passwordRepeat,
+        emailTooltip,
+        passwordTooltip,
+        passwordRepeatTooltip,
+        termsAccepted,
+        setTermsAccepted,
+        onTextChanged
+      }}>
+      {props.children}
+    </RegistrationContext.Provider>
+  );
```

Comment:
Tu jest tak w sumie bardzo dużo tego, sporo z tego jest używane tylko w jednym komponencie - moze tam powinno to zostać, no i jak jest za dużo wszystkiego na raz to imo traci na czytelności